generated/nidcpower/nidcpower/session.py

```diff
@@ -4801,11 +4801,39 @@ def fetch_multiple(self, count, timeout=hightime.timedelta(seconds=1.0)):
 
         '''
         import collections
-        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance'])
+        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance', 'channel'])
 
         voltage_measurements, current_measurements, in_compliance = self._fetch_multiple(timeout, count)
 
-        return [Measurement(voltage=voltage_measurements[i], current=current_measurements[i], in_compliance=in_compliance[i]) for i in range(count)]
+        if self._repeated_capability == '':
+            channel_names = self._get_channel_names(range(self.channel_count))
+        else:
+            # Check if the session was opened with independent_channels set to True by checking if
+            #  _get_channel_name() returns channel names with prefix
+            first_channel_name = self._get_channel_name(1)
```

Comment:
This seems brittle. You can get the value of `independent_channels` from `self._param_list`. Instead of parsing `self._param_list`, modify the constructor to store the session params as a dict and convert it to the list when needed (inside `__repr__`).

This should be its own PR.

Reply:
I've created #1755 to address this.

Reply:
@sbethur It seems that `independent_channels` is only available in `Session` and not `_SessionBase`, hence this approach of getting `independent_channels` from `self._param_list` would not work.

Also, even though the assumption that session opened with `independent_channels=True` would return channel name with prefix seems brittle, the actual conversion logic wouldn't work if this assumption is not valid, since it need to extract the prefix from the channel name and add it to channels repeated capability groups that are without prefix (again, the check here relies on the assumption that the prefix ends with a '/'). Hence, I propose keeping the current logic (it now lives in `convert_channels_repeated_capabilities()` following my latest commit).